File: src/Problem1TwoSum/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}  # {value: index}

        for idx, number in enumerate(nums):
            if target - number in hash:
                return [hash[target - number], idx]
            else:
                hash[number] = idx
        return


nums = [2, 7, 11, 15]
target = 9
print(Solution().twoSum(nums, target))

nums = [3, 2, 4]
target = 6
print(Solution().twoSum(nums, target))

nums = [3, 3]
target = 6
print(Solution().twoSum(nums, target))

File: src/Problem1TwoSum/TwoSum.java
package Problem1TwoSum;

import java.util.HashMap;

public class TwoSum {
    public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            if (hm.containsKey(target - nums[i])) {
                int[] ans = {i, hm.get(target - nums[i])};
                return ans;
            } else {
                hm.put(nums[i], i);
            }
        }
        return new int[2];
    }
    public static void main(String[] args) {
        int[] result = twoSum(new int[]{2,7,11,15}, 9);
        System.out.println("[" + result[0] + ", " + result[1] + "]");
    }

}
